lib/child_process.js

```diff
@@ -245,39 +243,29 @@ exports.execFile = function(file /*, args, options, callback*/) {
   }
 
   if (child.stdout) {
-    if (encoding)
-      child.stdout.setEncoding(encoding);
-
     child.stdout.addListener('data', function(chunk) {
-      stdoutLen += chunk.length;
+      stdoutLen += (chunk.byteLength || chunk.length);
```

Comment:
Yes, it does, but there's no avoiding that it could be a string if someone does something like this:

```
var child = child_process.spawn(...);
child.stdout.setEncoding('utf8');
```